### PR TITLE
Feature/sign in

### PR DESCRIPTION
**Pull request by lpatel99**
Created a sign in UI and implemented Google sign in with Firebase.

### CODE REVIEW

**Comment by lpatel99** [comment]
The low test coverage is normal, I use the Google API which I cannot really test with unit tests.

File: app/src/main/java/com/sdp13epfl2021/projmag/SignInActivity.kt [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+            val task = GoogleSignIn.getSignedInAccountFromIntent(data)
+            try {
+                // Google Sign In was successful, authenticate with Firebase
+                val account = task.getResult(ApiException::class.java)!!
+                Log.d("SignInActivity", "firebaseAuthWithGoogle:" + account.id)
+                firebaseAuthWithGoogle(account.idToken!!)
+            } catch (e: ApiException) {
+                // Google Sign In failed, update UI appropriately
+                Log.w("SignInActivity", "Google sign in failed", e)
+            }
```
Comment (arnauds5) [inline app/src/main/java/com/sdp13epfl2021/projmag/SignInActivity.kt]: I got the following error when running this code and when I clicked on my epfl account.
> com.sdp13epfl2021.projmag W/SignInActivity: Google sign in failed
>     com.google.android.gms.common.api.ApiException: 10: 
>         at com.google.android.gms.common.internal.ApiExceptionUtil.fromStatus(com.google.android.gms:play-services-base@@17.1.0:4)
>         at com.google.android.gms.auth.api.signin.GoogleSignIn.getSignedInAccountFromIntent(com.google.android.gms:play-services-auth@@19.0.0:9)
>         at com.sdp13epfl2021.projmag.SignInActivity.onActivityResult(SignInActivity.kt:55)
>         ...

Comment (lpatel99) [reply app/src/main/java/com/sdp13epfl2021/projmag/SignInActivity.kt]: The problem has been solved, it is no longer possible to click twice on the sign in button.

**Review by asfoury — COMMENTED** [review]
Handlers parameterless constructor is deprecated and may cause some problems, An alternative would be : 
https://stackoverflow.com/questions/61023968/what-do-i-use-now-that-handler-is-deprecated
Also after I signed in I stayed on the sign in page.

File: app/src/main/java/com/sdp13epfl2021/projmag/SignInActivity.kt [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        mAuth = FirebaseAuth.getInstance()
+
+        val gso = GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                .requestIdToken(getString(R.string.default_web_client_id))
+                .requestEmail()
+                .build()
+
+        googleSignInClient = GoogleSignIn.getClient(this, gso)
+
+        mAuth = FirebaseAuth.getInstance()
```
Comment (arnauds5) [inline app/src/main/java/com/sdp13epfl2021/projmag/SignInActivity.kt]: dupplicate line (line 35)

File: app/src/main/java/com/sdp13epfl2021/projmag/SignInActivity.kt [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        if (requestCode == RC_SIGN_IN) {
+            val task = GoogleSignIn.getSignedInAccountFromIntent(data)
+            try {
+                // Google Sign In was successful, authenticate with Firebase
+                val account = task.getResult(ApiException::class.java)!!
+                Log.d("SignInActivity", "firebaseAuthWithGoogle:" + account.id)
+                firebaseAuthWithGoogle(account.idToken!!)
+            } catch (e: ApiException) {
+                // Google Sign In failed, update UI appropriately
+                Log.w("SignInActivity", "Google sign in failed", e)
```
Comment (PaulCoral) [inline app/src/main/java/com/sdp13epfl2021/projmag/SignInActivity.kt:69]: Maybe use a [Toast](https://developer.android.com/guide/topics/ui/notifiers/toasts) message which will be visible to the user

**Review by PaulCoral — CHANGES_REQUESTED** [review]
use a Toast message instead of / with `Log.w(...)`

File: app/src/main/java/com/sdp13epfl2021/projmag/SignInActivity.kt [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        // Result returned from launching the Intent from GoogleSignInApi.getSignInIntent(...);
+        if (requestCode == RC_SIGN_IN) {
+            val task = GoogleSignIn.getSignedInAccountFromIntent(data)
+            try {
+                // Google Sign In was successful, authenticate with Firebase
+                val account = task.getResult(ApiException::class.java)!!
+                Log.d("SignInActivity", "firebaseAuthWithGoogle:" + account.id)
+                firebaseAuthWithGoogle(account.idToken!!)
+            } catch (e: ApiException) {
+                Toast.makeText(applicationContext, "Google sign in failed", Toast.LENGTH_SHORT)
```
Comment (PaulCoral) [inline app/src/main/java/com/sdp13epfl2021/projmag/SignInActivity.kt]: Warning : 
> Toast created but not shown: did you forget to call `show()` ?

**Review by PaulCoral — CHANGES_REQUESTED** [review]

**Review by PaulCoral — APPROVED** [review]
Looks good for me

**Review by arnauds5 — APPROVED** [review]
Looks good

**Pull request merged**